app/main.py

```diff
@@ -1,2 +1,56 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+
+    class Node:
+        def __init__(self, key: Any, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.hash = hash(key)
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.hash_table = [[] for _ in range(self.capacity)]
+
+    def __len__(self) -> int:
+        return self.size
+
+    def index(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def resize(self) -> None:
+        new_capacity = self.capacity * 2
+        new_hash_table = [[] for _ in range(new_capacity)]
+
+        for bucket in self.hash_table:
+            for node in bucket:
+                index = node.hash % new_capacity
+                new_hash_table[index].append(node)
+
+        self.hash_table = new_hash_table
+        self.capacity = new_capacity
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if self.size / self.capacity > self.load_factor:
```

Comment:
The condition for resizing the hash table is incorrect. It should be 'if self.size >= self.capacity * self.load_factor:' to ensure resizing occurs when the size exceeds the product of the capacity and the load factor, not when the ratio of size to capacity exceeds the load factor.